### ISSUE TITLE
Disabled time does not have aria-disabled attribute (unlike disabled dates)

### ISSUE THREAD

**Issue by alexkuc** [Hacker0x01/react-datepicker#4259]
**Describe the bug**
When certain time is excluded, it does _not_ have aria attribute `aria-disabled` unlike disabled dates

**To Reproduce**

1. Go to https://reactdatepicker.com/#example-exclude-times
2. Inspect `Time` column using Chrome's dev tools
3. See html output similar to this:
```html
<li class="react-datepicker__time-list-item  react-datepicker__time-list-item--disabled" tabindex="-1" role="option">5:00 PM</li>
```
4. Go to https://reactdatepicker.com/#example-exclude-dates
5. Inspect disabled date cell using Chrome's dev tools
6. See html output similar to this:
```html
<div class="react-datepicker__day react-datepicker__day--015 react-datepicker__day--disabled react-datepicker__day--excluded react-datepicker__day--selected react-datepicker__day--today" tabindex="0" aria-label="Not available Friday, September 15th, 2023" role="option" title="" aria-disabled="true" aria-current="date" aria-selected="true">15</div>
```

**Expected behavior**
See attribute `aria-disabled="true"` present for _both_, disabled dates _and_ disabled times.

**Comment by alexkuc** [comment]
If my understanding is correct, we only need to add aria attribute here:

https://github.com/Hacker0x01/react-datepicker/blob/9c249604050009e6875e614bbfd02bed646a5fe3/src/time.jsx#L208-L210

If this sounds right, I can submit a PR for this.

**Comment by martijnrusschen** [comment]
Go ahead, I can review a PR

**Comment by balajis-qb** [comment]
Hi @martijnrusschen, I tried fixing this issue along with the RTL test case migrate in my PR - https://github.com/Hacker0x01/react-datepicker/pull/4319.  Kindly review it and let me know if any changes required.
